src/oneapitest/Makefile

```diff
@@ -0,0 +1,117 @@
+TARGET_DIR := $(shell dirname $(realpath $(lastword $(MAKEFILE_LIST))))
+TARGET_NAME := $(notdir $(TARGET_DIR))
+TARGET := $(BASE_DIR)/$(TARGET_NAME)
+include Makefile.deps
+EXTERNAL_DEPENDS := $(oneapitest_EXTERNAL_DEPENDS)
+
+$(TARGET):
+test_cpu:
+test_sycl: $(TARGET)
```

Comment:
Just to clarify, the intention is that the suffix corresponds a platform to be tested, e.g. `cpu` for all CPU backends (like `serial` or `tbb` with Kokkos or Alpaka), `cuda` for NVIDIA GPUs etc.

Reply:
I didn't intentionally make any changes here - think I simply copied this from `cudatest` and did a global replace for `cuda` wih `sycl`...

What should the `Makefile` look like ?

Keep in mind that with `dpcpp` (or rather, with the open source `clang` fork) we would build a single binary, and choose the backend type (host, opencl, cuda, etc.) and device (cpu, gpu, etc.) at runtime,

Reply:
The runtime choice has been the goal for `kokkos` and `alpaka` as well. Both are basically running the same binary,  but with command line arguments specifying which backends to use. See e.g. for alpaka
https://github.com/cms-patatrack/pixeltrack-standalone/blob/c2e2642c9ccd1209626aba72869df1e39bb6a9a0/src/alpakatest/Makefile#L8-L18

For unit tests the approach needs to differ from `kokkos` and `alpaka`. For them, a separate binary is built for each backend. I suppose we could use command line arguments there as well for the single binary of SYCL.

Reply:
If needed, we can also force the selection of a specific backend and device using a dedicated device selector and compilation flags (or tie the selector to command line arguments).